app/controllers/taskmanagements_controller.rb

```diff
@@ -0,0 +1,78 @@
+class TaskmanagementsController < ApplicationController
+  before_action :show_notification_count, only: :new
+
+  def new
+    if params.except(:controller, :action).empty?
+      redirect_to(dashboard_path) && return
+    end
+
+    @taskee_id = deobfuscate(params.except(:controller, :action))["id"]
+    @taskee = User.find @taskee_id if @taskee_id
+    @task = TaskManagement.new
+  end
+
+  def create
+    @task = TaskManagement.new
+    @task.task_id = Task.find_by(name: task_details[:task_name].capitalize).id
```

Comment:
What do you think of passing task_details(`task_details.except(:task_name)`) as parameter to `TaskManagement.new`. This way, we won't need line 17 - 20.